webapp/graphite/finders/__init__.py

```diff
@@ -41,21 +42,37 @@ def extract_variants(pattern):
 
 
 def match_entries(entries, pattern):
-  """A drop-in replacement for fnmatch.filter that supports pattern
-  variants (ie. {foo,bar}baz = foobaz or barbaz)."""
-  v1, v2 = pattern.find('{'), pattern.find('}')
-
-  if v1 > -1 and v2 > v1:
-    variations = pattern[v1+1:v2].split(',')
-    variants = [ pattern[:v1] + v + pattern[v2+1:] for v in variations ]
+    # First we check for pattern variants (ie. {foo,bar}baz = foobaz or barbaz)
     matching = []
 
-    for variant in variants:
-      matching.extend( fnmatch.filter(entries, variant) )
-
-    return list( _deduplicate(matching) ) #remove dupes without changing order
-
-  else:
-    matching = fnmatch.filter(entries, pattern)
-    matching.sort()
-    return matching
+    for variant in expand_braces(pattern):
+      matching.extend(fnmatch.filter(entries, variant))
+
+    return list(_deduplicate(matching))
+
+
+"""
+    Brace expanding patch for python3 borrowed from:
+    https://bugs.python.org/issue9584
+"""
+def expand_braces(orig):
+    r = r'.*(\{.+?[^\\]\})'
+    p = re.compile(r)
+
+    s = orig[:]
```

Comment:
I don't think this copy is needed, `.replace` doesn't change the contents inplace afaics. (Just a couple of notes for optimization :-)